docs/DeveloperGuide.md

```diff
@@ -205,6 +205,55 @@ name, remark and product. Taking the aforementioned points into consideration, o
 implement the `find` command by taking in prefixes and throwing our relevant exceptions at appropriate points after
 considering code quality and end user experience.
 
+### Add supplier/warehouse feature
+The add supplier/warehouse mechanism is facilitated by the `AddCommandParser` and the `AddCommand`.
+The `AddCommandParser` implements `Parser` and the `AddCommand` extends `Command`, allowing the user to 
+add a supplier/warehouse to the app.
+
+#### What Add feature does
+The add feature allows users to add supplier/warehouse name, phone number and remarks. In addition, users
+can add email for suppliers and address for warehouses. Each supplier/warehouse is initialized without products associated to them.
+To add products, the addp feature should be invoked instead. Note that users are only able to add either
+a supplier or warehouse at any one time and not both or multiple at the same time.
+
+#### How it is implemented
+Step 1. After the `add` command is called, the user input will be sent to **AddCommandParser** for parsing.
+
+Step 2. **AddCommandParser** will then check if the compulsory prefixes `ct/COMMAND_TYPE n/NAME p/PHONE_NUMBER +
+addr/ADDRESS (for warehouse only)` is present. If the user enters any of the prefixes more than
```

Comment:
perhaps the "(for warehouse only)" shouldnt be within the tildas, same issue repeated below. Also is there a reason for the "+".

Reply:
I've broke it up to two parts so the warehouse only doesn't cover the front few prefixes.